pynitrokey/cli/nk3/test.py

```diff
@@ -234,6 +239,213 @@ def select(conn: CardConnection, aid: list[int]) -> bool:
+    "GenerateP521",
+    "EcDsaP521",
+    "VerifyP521",
+    "DeleteP521",
+    "RecreationWriteUserId",
+    "RecreationWriteBinary",
+    "RecreationDeleteAttempt",
+    "RecreationDeleteUserId",
+    "RecreationRecreateUserId",
+    "RecreationCreateSession",
+    "RecreationAuthSession",
+    "RecreationDeleteAttack",
+    "Rsa2048Gen",
+    "Rsa2048Sign",
+    "Rsa2048Verify",
+    "Rsa2048Encrypt",
+    "Rsa2048Decrypt",
+    "Rsa2048Delete",
+    "Rsa3072Gen",
+    "Rsa3072Sign",
+    "Rsa3072Verify",
+    "Rsa3072Encrypt",
+    "Rsa3072Decrypt",
+    "Rsa3072Delete",
+    "Rsa4096Gen",
+    "Rsa4096Sign",
+    "Rsa4096Verify",
+    "Rsa4096Encrypt",
+    "Rsa4096Decrypt",
+    "Rsa4096Delete",
+    "SymmWrite",
+    "SymmEncryptOneShot",
+    "SymmDecryptOneShot",
+    "SymmEncryptCreate",
+    "SymmEncryptInit",
+    "SymmEncryptUpdate1",
+    "SymmEncryptUpdate2",
+    "SymmEncryptFinal",
+    "SymmEncryptDelete",
+    "SymmDecryptCreate",
+    "SymmDecryptInit",
+    "SymmDecryptUpdate1",
+    "SymmDecryptUpdate2",
+    "SymmDecryptFinal",
+    "SymmDecryptDelete",
+    "SymmDelete",
+    "MacWrite",
+    "MacSignOneShot",
+    "MacVerifyOneShot",
+    "MacSignCreate",
+    "MacSignInit",
+    "MacSignUpdate1",
+    "MacSignUpdate2",
+    "MacSignFinal",
+    "MacSignDelete",
+    "MacVerifyCreate",
+    "MacVerifyInit",
+    "MacVerifyUpdate1",
+    "MacVerifyUpdate2",
+    "MacVerifyFinal",
+    "MacVerifyDelete",
+    "MacDelete",
+    "AesSessionCreateKey",
+    "AesSessionCreateBinary",
+    "AesSessionCreateSession",
+    "AesSessionAuthenticate",
+    "AesSessionReadBinary",
+    "AesSessionUpdateKey",
+    "AesSessionCloseSession",
+    "AesSessionRecreateSession",
+    "AesSessionReAuthenticate",
+    "AesSessionReadBinary2",
+    "AesSessionDeleteBinary",
+    "AesSessionDeleteKey",
+    "Pbkdf2WritePin",
+    "Pbkdf2Calculate",
+    "Pbkdf2Compare",
+    "Pbkdf2DeletePin",
+    "ImportWrite",
+    "ImportCipher",
+    "ImportExport",
+    "ImportDelete",
+    "ImportDeletionWorked",
+    "ImportImport",
+    "ImportCipher2",
+    "ImportComp",
+    "ImportDeleteFinal",
+]
+
+
+class NotSupported:
+    pass
+
+
+@test_case("se050", "SE050")
+def test_se050(ctx: TestContext, device: Nitrokey3Base) -> TestResult:
+    from queue import Queue
+
+    if not isinstance(device, Nitrokey3Device):
+        return TestResult(TestStatus.SKIPPED)
+    firmware_version = ctx.firmware_version or device.version()
+    if (
+        firmware_version.core() < Version(1, 5, 0)
+        or firmware_version.core() >= Version(1, 6, 0)
+        or firmware_version.pre is None
+    ):
+        return TestResult(TestStatus.SKIPPED)
+
+    que: Queue[Union[bytes, None, NotSupported]] = Queue()
+
+    def internal_se050_run(
+        q: Queue[Union[bytes, None, NotSupported]],
+    ) -> None:
+        try:
+            q.put(AdminApp(device).se050_tests())
+        except CtapError as e:
+            if e.code == CtapError.ERR.INVALID_LENGTH:
```

Comment:
We could also check for `INVALID_COMMAND` instead of whitelisting firmware versions.

Reply:
Returning `UnsupportedCommand` seems to lead to pynitrokey not receiving the error.

Reply:
Do you know why that is there: https://github.com/Nitrokey/pynitrokey/blob/master/pynitrokey/nk3/admin_app.py#L72

Reply:
I think the idea was to make it easier to handle the case that the firmware does not (yet) support a command.  `_call` always returns `bytes`, so `None` always indicates that the command is not supported.  And checking for `None` is easier than catching the exception and checking the error code.

Reply:
Ok, so I can just skip if None.

Reply:
Relevant PR in the admin app: https://github.com/Nitrokey/admin-app/pull/12